src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -0,0 +1,205 @@
+        else:
+            os_image_name = "MicrosoftWindowsServer:WindowsServer:2016-Datacenter:2016.127.20190214"
+
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    copied_os_disk_name = vm_name + '-DiskCopy-' + timestamp
+    copied_disk_uri = None
+    rescue_vm_name = ('rescue-' + vm_name)[:15]
+    rescue_rg_name = 'rescue-' + vm_name + '-' + timestamp
+
+    resource_tag = "rescue_source={rg}/{vm_name}".format(rg=resource_group_name, vm_name=vm_name)
+
+    # Set up base create vm command
+    create_rescue_vm_command = 'az vm create -g {g} -n {n} --tag {tag} --image {image} --admin-password {password}' \
+                               .format(g=rescue_rg_name, n=rescue_vm_name, tag=resource_tag, image=os_image_name, password=rescue_password)    
+    # Add username field only for Windows
+    if not is_linux:
+        create_rescue_vm_command += ' --admin-username {username}'.format(username=rescue_username)
+
+    # Main command calling block
+    try:
+        # fetch VM size of rescue VM
+        sku = _fetch_compatible_sku(target_vm)
+        if not sku:
+            raise Exception('Failed to find compatible VM size for target VM OS disk within given region and subscription.')
+        create_rescue_vm_command += ' --size {sku}'.format(sku=sku)
+
+        # Create New Resource Group
+        create_resource_group_command = 'az group create -l {loc} -n {group_name}' \
+                                        .format(loc=target_vm.location, group_name=rescue_rg_name)
+        logger.info('Creating resource group for rescue VM and its resources:')
+        _call_az_command(create_resource_group_command)
+
+        # MANAGED DISK
+        if is_managed:
+            logger.info('OS disk is managed. Executing managed disk swap:\n')
+            # Copy OS disk command
+            copy_disk_command = 'az disk create -g {g} -n {n} --source {s} --query id -o tsv' \
+                                .format(g=resource_group_name, n=copied_os_disk_name, s=target_disk_name)
+            # Validate create vm create command to validate parameters before runnning copy disk command
+            validate_create_vm_command = create_rescue_vm_command + ' --validate'
+
+            logger.info('Validating VM template before continuing:')
+            _call_az_command(validate_create_vm_command)
+            logger.info('Copying OS disk of target VM:')
+            copy_disk_id = _call_az_command(copy_disk_command)
+
+            attach_disk_command = 'az vm disk attach -g {g} --vm-name {rescue} --name {id}' \
+                                  .format(g=rescue_rg_name, rescue=rescue_vm_name, id=copy_disk_id)
+
+            logger.info('Creating rescue vm:')
+            _call_az_command(create_rescue_vm_command)
+            logger.info('Attaching copied disk to rescue vm:')
+            _call_az_command(attach_disk_command)
+        # UNMANAGED DISK
```

Comment:
Just something to note down: not sure about if we need a scenario for premium disk and standard disk?